appserver/ejb/ejb-http-remoting/client/src/main/java/fish/payara/ejb/http/client/LookupDiscovery.java

```diff
@@ -0,0 +1,164 @@
+ *    See the License for the specific
+ *    language governing permissions and limitations under the License.
+ *
+ *    When distributing the software, include this License Header Notice in each
+ *    file and include the License file at glassfish/legal/LICENSE.txt.
+ *
+ *    GPL Classpath Exception:
+ *    The Payara Foundation designates this particular file as subject to the "Classpath"
+ *    exception as provided by the Payara Foundation in the GPL Version 2 section of the License
+ *    file that accompanied this code.
+ *
+ *    Modifications:
+ *    If applicable, add the following below the License Header, with the fields
+ *    enclosed by brackets [] replaced by your own identifying information:
+ *    "Portions Copyright [year] [name of copyright owner]"
+ *
+ *    Contributor(s):
+ *    If you wish your version of this file to be governed by only the CDDL or
+ *    only the GPL Version 2, indicate your decision by adding "[Contributor]
+ *    elects to include this software in this distribution under the [CDDL or GPL
+ *    Version 2] license."  If you don't indicate a single choice of license, a
+ *    recipient has the option to distribute your version of this file under
+ *    either the CDDL, the GPL Version 2 or to extend the choice of license to
+ *    its licensees as provided above.  However, if you add GPL Version 2 code
+ *    and therefore, elected the GPL Version 2 license, then the option applies
+ *    only if the new code is made subject to such option by the copyright
+ *    holder.
+ */
+
+package fish.payara.ejb.http.client;
+
+import javax.naming.NamingException;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Link;
+import javax.ws.rs.core.Response;
+import java.net.URI;
+import java.util.Map;
+import java.util.Set;
+
+import static javax.ws.rs.core.Response.Status.Family.CLIENT_ERROR;
+import static javax.ws.rs.core.Response.Status.Family.REDIRECTION;
+import static javax.ws.rs.core.Response.Status.Family.SERVER_ERROR;
+import static javax.ws.rs.core.Response.Status.Family.SUCCESSFUL;
+import static javax.ws.rs.core.Response.Status.NOT_FOUND;
+import static javax.ws.rs.core.Response.Status.OK;
+
+class LookupDiscovery {
+    static final String INVOKER_V1_REL = "<https://payara.fish/ejb-http-invoker/v1>";
+    static final String INVOKER_V0_REL = "<https://payara.fish/ejb-http-invoker/v0>";
+
+    private final Map<String, Object> environment;
+    private final URI root;
+    private final Client client;
+    private URI resolvedRoot;
+    private boolean v0target;
+    private WebTarget v0lookup;
+    private boolean v1target;
+    private WebTarget v1lookup;
+
+    LookupDiscovery(Map<String, Object> environment, Client client, URI root) {
+        this.environment = environment;
+        this.root = root;
+        this.resolvedRoot = root;
+        this.client = client;
+    }
+
+    void discover() throws NamingException {
```

Comment:
To me this should return the result of the discovery as data which allows the caller to inspect.